grails-app/domain/org/pih/warehouse/invoice/InvoiceItemType.groovy

```diff
@@ -0,0 +1,14 @@
+package org.pih.warehouse.invoice
+
+enum InvoiceItemType {
+
+    INVERSE('Inverse'),
```

Comment:
@jmiranda Let me know if other types come to your mind you'd like to have added here

Reply:
If we added InvoiceTypeCode.FINAL, could the INVERSEness of an item be derived?  I assume that the other two (prepayment, regular) could already be derived from the InvoiceType. So even if there's a need to store whether an invoice item is an "inverse" item, we could do so with just a boolean field.

Side note: If/when you add REGULAR InvoiceTypeCode it should be called STANDARD.

Source: https://www.freshbooks.com/hub/invoicing/types-of-invoices

Reply:
@jmiranda Yeah, to be honest, this part was not designed properly, because we wanted to go with calculating all the time and we switched to keeping inverse items on the final invoice right before we started the development. So something that was my **suggested direction** in case we want to go with storing inverse items somehow was cemented in my brain as the version we want to do. As deriving item type from invoice type does not work for me here (due to mixing both standard/regular items with inverse items), I decided that the boolean fits here more, and this is what I implemented for now (we can still revise it again in the final feature PR).